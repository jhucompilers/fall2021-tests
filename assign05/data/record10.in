1 36 52 27 86 47 30 99 85 53 26 4 82 57 69 35
