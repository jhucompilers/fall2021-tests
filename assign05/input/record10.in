PROGRAM record10;
  TYPE Pair = RECORD
    x, y: INTEGER;
  END;
  VAR pairs : ARRAY 8 OF Pair;
  VAR i, tmp : INTEGER;

BEGIN
  -- read pairs from input
  i := 0;
  WHILE i < 8 DO
    READ pairs[i].x;
    READ pairs[i].y;
    i := i + 1;
  END;

  -- write pairs
  i := 0;
  WHILE i < 8 DO
    WRITE pairs[i].x;
    WRITE pairs[i].y;
    i := i + 1;
  END;

  -- swap x and y in each element
  i := 0;
  WHILE i < 8 DO
    tmp := pairs[i].x;
    pairs[i].x := pairs[i].y;
    pairs[i].y := tmp;
    i := i + 1;
  END;

  -- write pairs again
  i := 0;
  WHILE i < 8 DO
    WRITE pairs[i].x;
    WRITE pairs[i].y;
    i := i + 1;
  END;
END.
