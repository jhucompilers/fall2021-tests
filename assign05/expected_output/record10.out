1
36
52
27
86
47
30
99
85
53
26
4
82
57
69
35
36
1
27
52
47
86
99
30
53
85
4
26
57
82
35
69
